scripts/install

```diff
@@ -29,233 +26,162 @@ path_url=$YNH_APP_ARG_PATH
-final_path=/var/www/$app
-test ! -e "$final_path" || ynh_die "This path already contains a folder"
-
-# Normalize the url path syntax
-path_url=$(ynh_normalize_url_path $path_url)
+ynh_print_info --message="Validating installation parameters..."
 
 # Register (book) web path
-ynh_webpath_register $app $domain $path_url
+ynh_webpath_register --app=$app --domain=$domain --path_url=$path_url
 
 #=================================================
 # STORE SETTINGS FROM MANIFEST
 #=================================================
-ynh_print_info "Storing installation settings..."
+ynh_print_info --message="Storing installation settings..."
 
-ynh_app_setting_set $app domain $domain
-ynh_app_setting_set $app path $path_url
-ynh_app_setting_set $app is_public $is_public
-ynh_app_setting_set $app nextclouddomain $nextclouddomain
+ynh_app_setting_set --app=$app --key=domain --value=$domain
+ynh_app_setting_set --app=$app --key=path --value=$path_url
+ynh_app_setting_set --app=$app --key=is_public --value=$is_public
+ynh_app_setting_set --app=$app --key=nextclouddomain --value=$nextclouddomain
 
 #=================================================
 # STANDARD MODIFICATIONS
 #=================================================
 # FIND AND OPEN A PORT
 #=================================================
-ynh_print_info "Configuring firewall..."
-
-### Use these lines if you have to open a port for the application
-### `ynh_find_port` will find the first available port starting from the given port.
-### If you're not using these lines:
-###		- Remove the section "CLOSE A PORT" in the remove script
+ynh_print_info --message="Configuring firewall..."
 
 # Find a free port
-port=$(ynh_find_port 8095)
-# Open this port
-#ynh_exec_warn_less yunohost firewall allow --no-upnp TCP $port
-ynh_app_setting_set $app port $port
+port=$(ynh_find_port --port=8095)
+ynh_app_setting_set --app=$app --key=port --value=$port
 
 #=================================================
 # ADD NODEJS REPOSITORY
 #=================================================
-ynh_print_info "Add nodejs repository..."
+ynh_print_info --message="Add nodejs repository..."
 
-curl -sL https://deb.nodesource.com/setup_8.x | sudo bash -
+ynh_install_extra_repo --repo="deb https://deb.nodesource.com/node_8.x stretch main" --key="https://deb.nodesource.com/gpgkey/nodesource.gpg.key"
 
 #=================================================
 # ADD ONLYOFFICE REPOSITORY
 #=================================================
-ynh_print_info "Add OnlyOffice repository..."
+ynh_print_info --message="Add OnlyOffice repository..."
 
 apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys CB2DE8E5
-echo "deb http://download.onlyoffice.com/repo/debian squeeze main" | sudo tee /etc/apt/sources.list.d/onlyoffice.list
+ynh_install_extra_repo --repo="deb http://download.onlyoffice.com/repo/debian squeeze main" --append
```

Comment:
I guess this is for the installation of onlyoffice-documentserver, later into that script.

It's better to use [`ynh_install_extra_app_dependencies`](https://github.com/YunoHost-Apps/Experimental_helpers/blob/master/ynh_add_extra_apt_repos/ynh_add_extra_apt_repos__3#L184) as this helper will add the repo, install the package then remove the repo.
So it won't have any effect on apt later.

Reply:
when using
```
ynh_install_extra_app_dependencies --repo="deb http://download.onlyoffice.com/repo/debian squeeze main" --package="$pkg_dependencies onlyoffice-documentserver"
```
or
```
ynh_install_extra_app_dependencies --repo="deb http://download.onlyoffice.com/repo/debian squeeze main" --package="$pkg_dependencies"
ynh_install_extra_app_dependencies --repo="deb http://download.onlyoffice.com/repo/debian squeeze main" --package="onlyoffice-documentserver"
```
the installation fails with error:
```
128194 DEBUG The following packages were automatically installed and are no longer required:
128194 DEBUG   erlang-asn1 erlang-base erlang-corba erlang-crypto erlang-diameter
128194 DEBUG   erlang-edoc erlang-eldap erlang-erl-docgen erlang-eunit erlang-ic
128195 DEBUG   erlang-inets erlang-mnesia erlang-nox erlang-odbc erlang-os-mon
128195 DEBUG   erlang-parsetools erlang-percept erlang-public-key erlang-runtime-tools
128196 DEBUG   erlang-snmp erlang-ssh erlang-ssl erlang-syntax-tools erlang-tools
128196 DEBUG   erlang-xmerl libcurl3-nss libcurl4-nss-dev libnspr4 libnss3 libsctp1
128196 DEBUG   postgresql rabbitmq-server
128197 DEBUG Use 'apt autoremove' to remove them.
128297 DEBUG The following packages will be REMOVED:
128298 WARNING E: Packages need to be removed but remove is disabled.
128298 DEBUG   onlyoffice-ynh-deps
128299 WARNING Unable to install dependencies
128299 DEBUG 0 upgraded, 0 newly installed, 1 to remove and 2 not upgraded.
128299 WARNING !!
128300 DEBUG + ynh_die '--message=Unable to install dependencies'
128300 WARNING   onlyoffice's script has encountered an error. Its execution was cancelled.
```



So the only way I found to install was to use:

1. `ynh_install_extra_repo --repo="deb http://download.onlyoffice.com/repo/debian squeeze main" --append`
2. `ynh_add_app_dependencies --package="onlyoffice-documentserver"`

Reply:
If you have issues with the helper, as its an experimental helper, that would be better to try to fix it.

Reply:
lol, I know. Not sure it comes from the helper, took me hours to make the package works and not enough time to deep dive more into that package ^^